pallets/sponsorship/src/lib.rs

```diff
@@ -17,6 +17,7 @@
  */
 
 #![cfg_attr(not(feature = "std"), no_std)]
+#![deny(unused_must_use)]
```

Comment:
I suppose this is temporary for your own test. Otherwise, the warning is the preferred behaviour in the case of unused return value.

Reply:
ehhh